components/omega/test/ocn/EosTest.cpp

```diff
@@ -0,0 +1,540 @@
+   // the default MachEnv. Then retrieve the default environment and
+   // some needed data members.
+   MachEnv::init(MPI_COMM_WORLD);
+   MachEnv *DefEnv  = MachEnv::getDefault();
+   MPI_Comm DefComm = DefEnv->getComm();
+
+   initLogging(DefEnv);
+   LOG_INFO("… in initEosTest");
+
+   // Open config file
+   Config("Omega");
+   Err = Config::readAll("omega.yml");
+   if (Err != 0) {
+      LOG_ERROR("Eos: Error reading config file");
+      return Err;
+   }
+
+   int TimeStepperErr = TimeStepper::init1();
+   if (TimeStepperErr != 0) {
+      Err++;
+      LOG_ERROR("TendenciesTest: error initializing default time stepper");
+   }
+
+   int IOErr = IO::init(DefComm);
+   if (IOErr != 0) {
+      Err++;
+      LOG_ERROR("TendenciesTest: error initializing parallel IO");
+   }
+
+   int DecompErr = Decomp::init(mesh);
+   if (DecompErr != 0) {
+      Err++;
+      LOG_ERROR("TendenciesTest: error initializing default decomposition");
+   }
+
+   int HaloErr = Halo::init();
+   if (HaloErr != 0) {
+      Err++;
+      LOG_ERROR("TendenciesTest: error initializing default halo");
+   }
+
+   int MeshErr = HorzMesh::init();
+   if (MeshErr != 0) {
+      Err++;
+      LOG_ERROR("TendenciesTest: error initializing default mesh");
+   }
+
+   int TracerErr = Tracers::init();
+   if (TracerErr != 0) {
+      Err++;
+      LOG_ERROR("TendenciesTest: error initializing tracer infrastructure");
+   }
+
+   const auto &Mesh = HorzMesh::getDefault();
+   std::shared_ptr<Dimension> VertDim =
+       Dimension::create("NVertLevels", NVertLevels);
+
+   return Err;
+}
+
+int testEosMapping() {
+   int Err = 0;
+   // test initialization
+   int EosErr = Eos::init();
+   if (EosErr != 0) {
+      Err++;
+      LOG_ERROR("EosTest: error initializing default Eos");
+   }
+
+   // test retrievel of default
+   Eos *DefEos = Eos::getDefault();
+
+   if (DefEos) {
+      LOG_INFO("EosTest: Default Eos retrieval PASS");
+   } else {
+      Err++;
+      LOG_INFO("EosTest: Default Eos retrieval FAIL");
+      return -1;
+   }
+
+   const auto *Mesh = HorzMesh::getDefault();
+   // test creation of another Eos
+   Eos::create("TestEos", Mesh, NVertLevels);
+
+   if (Eos::get("TestEos")) {
+      LOG_INFO("EosTest: Non-default Eos retrieval PASS");
+   } else {
+      Err++;
+      LOG_INFO("EosTest: Non-default Eos retrieval FAIL");
+   }
+
+   // test erase
+   Eos::erase("TestEos");
+
+   if (Eos::get("TestEos")) {
+      Err++;
+      LOG_INFO("EosTest: Non-default Eos erase FAIL");
+   } else {
+      LOG_INFO("EosTest: Non-default Eos erase PASS");
+   }
+
+   // Eos::clear();
+
+   return Err;
+}
+
+int testEosLinear() {
+   int Err          = 0;
+   const auto *Mesh = HorzMesh::getDefault();
+   // create Eos to test
+   Eos::create("LinearEos", Mesh, NVertLevels);
+   Eos *TestEos       = Eos::get("LinearEos");
+   TestEos->eosChoice = EosType::Linear;
+   Real Threshold     = LinearExpValue;
```

Comment:
`Threshold` kind of sounds like a tolerance. Do we need a local variable to store the expected value?

Reply:
Agreed. I already have the value stored. It used to be the same block of code for both options so it was easier to have a local variable to toggle. Not needed now.